### ISSUE TITLE
Model have 0 accuracy when eval or using exported model but high accuracy in training

### ISSUE THREAD

**Issue by BaoNguyen6742** [PaddlePaddle/PaddleOCR#17275]
### 🔎 Search before asking

- [x] I have searched the PaddleOCR [Docs](https://paddlepaddle.github.io/PaddleOCR/) and found no similar bug report.
- [x] I have searched the PaddleOCR [Issues](https://github.com/PaddlePaddle/PaddleOCR/issues) and found no similar bug report.
- [x] I have searched the PaddleOCR [Discussions](https://github.com/PaddlePaddle/PaddleOCR/discussions) and found no similar bug report.

### 🐛 Bug (问题描述)

When I train the model, the accuracy is around 0.8 for many epochs, but when I export the model it's always 0 accuracy and predict garbage text. Also it predict nearly every images with the same garbage text. Even when I export the model it still have 0 accuracy when predict

This is my config

```
Global:
  model_name: PP-OCRv5_server_rec # To use static model for inference.
  debug: false
  use_gpu: true
  epoch_num: 1000
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/PP-OCRv5_server_rec
  save_epoch_step: 1
  eval_batch_step: [0, 2000]
  cal_metric_during_train: true
  calc_epoch_interval: 1
  pretrained_model: 
  checkpoints:
  save_inference_dir:
  use_visualdl: false
  infer_img:
  character_dict_path: Short_dataset/dict.txt
  max_text_length: &max_text_length 33
  infer_mode: false
  use_space_char: true
  distributed: true
  save_res_path: ./output/rec/predicts_ppocrv5.txt
  d2s_train_image_shape: [3, 50, 450]


Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.0005
    warmup_epoch: 1
  regularizer:
    name: L2
    factor: 3.0e-05


Architecture:
  model_type: rec
  algorithm: SVTR_HGNet
  Transform:
  Backbone:
    name: PPHGNetV2_B4
    text_rec: True
  Head:
    name: MultiHead
    head_list:
      - CTCHead:
          Neck:
            name: svtr
            dims: 120
            depth: 2
            hidden_dims: 120
            kernel_size: [1, 3]
            use_guide: True
          Head:
            fc_decay: 0.00001
      - NRTRHead:
          nrtr_dim: 384
          max_text_length: *max_text_length

Loss:
  name: MultiLoss
  loss_config_list:
    - CTCLoss:
    - NRTRLoss:

PostProcess:  
  name: CTCLabelDecode

Metric:
  name: RecMetric
  main_indicator: acc

Train:
  dataset:
    name: MultiScaleDataSet
    ds_width: false
    data_dir: Short_dataset/images
    ext_op_transform_idx: 1
    label_file_list:
    - Short_dataset/images/train.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - RecAug:
    - MultiLabelEncode:
        gtc_encode: NRTRLabelEncode
    - KeepKeys:
        keep_keys:
        - image
        - label_ctc
        - label_gtc
        - length
        - valid_ratio
  sampler:
    name: MultiScaleSampler
    scales: [[400, 50], [450, 50], [500, 50]]
    first_bs: &bs 128
    fix_bs: false
    divided_factor: [8, 16] # w, h
    is_training: True
  loader:
    shuffle: true
    batch_size_per_card: *bs
    drop_last: true
    num_workers: 16
Eval:
  dataset:
    name: SimpleDataSet
    data_dir: Short_dataset/images
    label_file_list:
    - Short_dataset/images/val.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - MultiLabelEncode:
        gtc_encode: NRTRLabelEncode
    - RecResizeImg:
        image_shape: [3, 50, 450]
    - KeepKeys:
        keep_keys:
        - image
        - label_ctc
        - label_gtc
        - length
        - valid_ratio
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 128
    num_workers: 4
```

The log when I train the model

```
[2025/12/02 07:06:37] ppocr INFO: epoch: [49/1000], global_step: 2170, lr: 0.000497, acc: 0.805555, norm_edit_dis: 0.983497, CTCLoss: 1.546005, NRTRLoss: 0.886353, loss: 2.443305, avg_reader_cost: 2.30333 s, avg_batch_cost: 3.96048 s, avg_samples: 90.0, ips: 22.72453 samples/s, eta: 22:55:44, max_mem_reserved: 10405 MB, max_mem_allocated: 8634 MB
[2025/12/02 07:06:52] ppocr INFO: epoch: [49/1000], global_step: 2180, lr: 0.000497, acc: 0.822222, norm_edit_dis: 0.985446, CTCLoss: 1.546005, NRTRLoss: 0.886682, loss: 2.443305, avg_reader_cost: 0.02050 s, avg_batch_cost: 1.48327 s, avg_samples: 90.0, ips: 60.67677 samples/s, eta: 22:53:58, max_mem_reserved: 10405 MB, max_mem_allocated: 8634 MB
[2025/12/02 07:07:05] ppocr INFO: epoch: [49/1000], global_step: 2190, lr: 0.000497, acc: 0.800000, norm_edit_dis: 0.983193, CTCLoss: 1.501329, NRTRLoss: 0.891330, loss: 2.391097, avg_reader_cost: 0.00578 s, avg_batch_cost: 1.35993 s, avg_samples: 90.0, ips: 66.17991 samples/s, eta: 22:51:48, max_mem_reserved: 10405 MB, max_mem_allocated: 8634 MB
[2025/12/02 07:07:18] ppocr INFO: epoch: [49/1000], global_step: 2200, lr: 0.000497, acc: 0.805555, norm_edit_dis: 0.982435, CTCLoss: 1.772196, NRTRLoss: 0.891632, loss: 2.666835, avg_reader_cost: 0.00424 s, avg_batch_cost: 1.27896 s, avg_samples: 90.0, ips: 70.36954 samples/s, eta: 22:49:24, max_mem_reserved: 10405 MB, max_mem_allocated: 8634 MB
[2025/12/02 07:07:24] ppocr INFO: epoch: [49/1000], global_step: 2205, lr: 0.000497, acc: 0.811111, norm_edit_dis: 0.983715, CTCLoss: 1.855664, NRTRLoss: 0.890527, loss: 2.747430, avg_reader_cost: 0.00192 s, avg_batch_cost: 0.62228 s, avg_samples: 45.0, ips: 72.31512 samples/s, eta: 22:48:09, max_mem_reserved: 10405 MB, max_mem_allocated: 8634 MB
```

The result when I run the eval
`python ./PaddleOCR/tools/eval.py -c "Short_dataset/PPOCR_config.yaml" -o Global.pretrain_model="Short_line_result/output/PP-OCRv5_server_rec/latest/latest.pdparams"`

```
Skipping import of the encryption module.
[2025/12/02 14:08:55] ppocr INFO: Architecture : 
[2025/12/02 14:08:55] ppocr INFO:     Backbone : 
[2025/12/02 14:08:55] ppocr INFO:         name : PPHGNetV2_B4
[2025/12/02 14:08:55] ppocr INFO:         text_rec : True
[2025/12/02 14:08:55] ppocr INFO:     Head : 
[2025/12/02 14:08:55] ppocr INFO:         head_list : 
[2025/12/02 14:08:55] ppocr INFO:             CTCHead : 
[2025/12/02 14:08:55] ppocr INFO:                 Head : 
[2025/12/02 14:08:55] ppocr INFO:                     fc_decay : 1e-05
[2025/12/02 14:08:55] ppocr INFO:                 Neck : 
[2025/12/02 14:08:55] ppocr INFO:                     depth : 2
[2025/12/02 14:08:55] ppocr INFO:                     dims : 120
[2025/12/02 14:08:55] ppocr INFO:                     hidden_dims : 120
[2025/12/02 14:08:55] ppocr INFO:                     kernel_size : [1, 3]
[2025/12/02 14:08:55] ppocr INFO:                     name : svtr
[2025/12/02 14:08:55] ppocr INFO:                     use_guide : True
[2025/12/02 14:08:55] ppocr INFO:             NRTRHead : 
[2025/12/02 14:08:55] ppocr INFO:                 max_text_length : 33
[2025/12/02 14:08:55] ppocr INFO:                 nrtr_dim : 384
[2025/12/02 14:08:55] ppocr INFO:         name : MultiHead
[2025/12/02 14:08:55] ppocr INFO:     Transform : None
[2025/12/02 14:08:55] ppocr INFO:     algorithm : SVTR_HGNet
[2025/12/02 14:08:55] ppocr INFO:     model_type : rec
[2025/12/02 14:08:55] ppocr INFO: Eval :
[2025/12/02 14:08:55] ppocr INFO:     dataset :
[2025/12/02 14:08:55] ppocr INFO:         data_dir : Short_dataset/images
[2025/12/02 14:08:55] ppocr INFO:         label_file_list : ['Short_dataset/images/val.txt']
[2025/12/02 14:08:55] ppocr INFO:         name : SimpleDataSet
[2025/12/02 14:08:55] ppocr INFO:         transforms :
[2025/12/02 14:08:55] ppocr INFO:             DecodeImage :
[2025/12/02 14:08:55] ppocr INFO:                 channel_first : False
[2025/12/02 14:08:55] ppocr INFO:                 img_mode : BGR
[2025/12/02 14:08:55] ppocr INFO:             MultiLabelEncode :
[2025/12/02 14:08:55] ppocr INFO:                 gtc_encode : NRTRLabelEncode
[2025/12/02 14:08:55] ppocr INFO:             RecResizeImg :
[2025/12/02 14:08:55] ppocr INFO:                 image_shape : [3, 50, 450]
[2025/12/02 14:08:55] ppocr INFO:             KeepKeys :
[2025/12/02 14:08:55] ppocr INFO:                 keep_keys : ['image', 'label_ctc', 'label_gtc', 'length', 'valid_ratio']
[2025/12/02 14:08:55] ppocr INFO:     loader :
[2025/12/02 14:08:55] ppocr INFO:         batch_size_per_card : 128
[2025/12/02 14:08:55] ppocr INFO:         drop_last : False
[2025/12/02 14:08:55] ppocr INFO:         num_workers : 4
[2025/12/02 14:08:55] ppocr INFO:         shuffle : False
[2025/12/02 14:08:55] ppocr INFO: Global :
[2025/12/02 14:08:55] ppocr INFO:     cal_metric_during_train : True
[2025/12/02 14:08:55] ppocr INFO:     calc_epoch_interval : 1
[2025/12/02 14:08:55] ppocr INFO:     character_dict_path : Short_dataset/dict.txt
[2025/12/02 14:08:55] ppocr INFO:     checkpoints : None
[2025/12/02 14:08:55] ppocr INFO:     d2s_train_image_shape : [3, 50, 450]
[2025/12/02 14:08:55] ppocr INFO:     debug : False
[2025/12/02 14:08:55] ppocr INFO:     distributed : False
[2025/12/02 14:08:55] ppocr INFO:     epoch_num : 75
[2025/12/02 14:08:55] ppocr INFO:     eval_batch_step : [0, 2000]
[2025/12/02 14:08:55] ppocr INFO:     infer_img : None
[2025/12/02 14:08:55] ppocr INFO:     infer_mode : False
[2025/12/02 14:08:55] ppocr INFO:     log_smooth_window : 20
[2025/12/02 14:08:55] ppocr INFO:     max_text_length : 33
[2025/12/02 14:08:55] ppocr INFO:     model_name : PP-OCRv5_server_rec
[2025/12/02 14:08:55] ppocr INFO:     pretrain_model : Short_line_result/output/PP-OCRv5_server_rec/latest/latest.pdparams
[2025/12/02 14:08:55] ppocr INFO:     pretrained_model : None
[2025/12/02 14:08:55] ppocr INFO:     print_batch_step : 10
[2025/12/02 14:08:55] ppocr INFO:     save_epoch_step : 1
[2025/12/02 14:08:55] ppocr INFO:     save_inference_dir : None
[2025/12/02 14:08:55] ppocr INFO:     save_model_dir : ./output/PP-OCRv5_server_rec
[2025/12/02 14:08:55] ppocr INFO:     save_res_path : ./output/rec/predicts_ppocrv5.txt
[2025/12/02 14:08:55] ppocr INFO:     use_gpu : True
[2025/12/02 14:08:55] ppocr INFO:     use_space_char : True
[2025/12/02 14:08:55] ppocr INFO:     use_visualdl : False
[2025/12/02 14:08:55] ppocr INFO: Loss :
[2025/12/02 14:08:55] ppocr INFO:     loss_config_list :
[2025/12/02 14:08:55] ppocr INFO:         CTCLoss : None
[2025/12/02 14:08:55] ppocr INFO:         NRTRLoss : None
[2025/12/02 14:08:55] ppocr INFO:     name : MultiLoss
[2025/12/02 14:08:55] ppocr INFO: Metric :
[2025/12/02 14:08:55] ppocr INFO:     main_indicator : acc
[2025/12/02 14:08:55] ppocr INFO:     name : RecMetric
[2025/12/02 14:08:55] ppocr INFO: Optimizer :
[2025/12/02 14:08:55] ppocr INFO:     beta1 : 0.9
[2025/12/02 14:08:55] ppocr INFO:     beta2 : 0.999
[2025/12/02 14:08:55] ppocr INFO:     lr :
[2025/12/02 14:08:55] ppocr INFO:         learning_rate : 0.0005
[2025/12/02 14:08:55] ppocr INFO:         name : Cosine
[2025/12/02 14:08:55] ppocr INFO:         warmup_epoch : 1
[2025/12/02 14:08:55] ppocr INFO:     name : Adam
[2025/12/02 14:08:55] ppocr INFO:     regularizer :
[2025/12/02 14:08:55] ppocr INFO:         factor : 3e-05
[2025/12/02 14:08:55] ppocr INFO:         name : L2
[2025/12/02 14:08:55] ppocr INFO: PostProcess :
[2025/12/02 14:08:55] ppocr INFO:     name : CTCLabelDecode
[2025/12/02 14:08:55] ppocr INFO: Train :
[2025/12/02 14:08:55] ppocr INFO:     dataset :
[2025/12/02 14:08:55] ppocr INFO:         data_dir : Short_dataset/images
[2025/12/02 14:08:55] ppocr INFO:         ds_width : False
[2025/12/02 14:08:55] ppocr INFO:         ext_op_transform_idx : 1
[2025/12/02 14:08:55] ppocr INFO:         label_file_list : ['Short_dataset/images/train.txt']       
[2025/12/02 14:08:55] ppocr INFO:         name : MultiScaleDataSet
[2025/12/02 14:08:55] ppocr INFO:         transforms :
[2025/12/02 14:08:55] ppocr INFO:             DecodeImage :
[2025/12/02 14:08:55] ppocr INFO:                 channel_first : False
[2025/12/02 14:08:55] ppocr INFO:                 img_mode : BGR
[2025/12/02 14:08:55] ppocr INFO:             RecAug : None
[2025/12/02 14:08:55] ppocr INFO:             MultiLabelEncode :
[2025/12/02 14:08:55] ppocr INFO:                 gtc_encode : NRTRLabelEncode
[2025/12/02 14:08:55] ppocr INFO:             KeepKeys :
[2025/12/02 14:08:55] ppocr INFO:                 keep_keys : ['image', 'label_ctc', 'label_gtc', 'length', 'valid_ratio']
[2025/12/02 14:08:55] ppocr INFO:     loader :
[2025/12/02 14:08:55] ppocr INFO:         batch_size_per_card : 128
[2025/12/02 14:08:55] ppocr INFO:         drop_last : True
[2025/12/02 14:08:55] ppocr INFO:         num_workers : 16
[2025/12/02 14:08:55] ppocr INFO:         shuffle : True
[2025/12/02 14:08:55] ppocr INFO:     sampler :
[2025/12/02 14:08:55] ppocr INFO:         divided_factor : [8, 16]
[2025/12/02 14:08:55] ppocr INFO:         first_bs : 128
[2025/12/02 14:08:55] ppocr INFO:         fix_bs : False
[2025/12/02 14:08:55] ppocr INFO:         is_training : True
[2025/12/02 14:08:55] ppocr INFO:         name : MultiScaleSampler
[2025/12/02 14:08:55] ppocr INFO:         scales : [[400, 50], [450, 50], [500, 50]]
[2025/12/02 14:08:55] ppocr INFO: profiler_options : None
[2025/12/02 14:08:55] ppocr INFO: train with paddle 3.2.2 and device Place(gpu:0)
[2025/12/02 14:08:55] ppocr INFO: Initialize indexes of datasets:['Short_dataset/images/val.txt']    
WARNING: Logging before InitGoogleLogging() is written to STDERR
W1202 14:08:55.560381 14216 gpu_resources.cc:114] Please NOTE: device: 0, GPU Compute Capability: 8.6, Driver API Version: 13.0, Runtime API Version: 12.6
[2025/12/02 14:08:55] ppocr INFO: train from scratch
[2025/12/02 14:08:55] ppocr INFO: metric in ckpt ***************
[2025/12/02 14:08:55] ppocr INFO: is_float16:False
eval model::  94%|████████████████████████████████████████████████████████████████████████████████████████▏     | 15/16 [01:10<00:04,  4.73s/it] 
[2025/12/02 14:10:06] ppocr INFO: metric eval ***************
[2025/12/02 14:10:06] ppocr INFO: acc:0.0
[2025/12/02 14:10:06] ppocr INFO: norm_edit_dis:0.01339420065065089
[2025/12/02 14:10:06] ppocr INFO: fps:27.157053688615335
```

### 🏃‍♂️ Environment (运行环境)

I'm running on Kaggle with paddlepaddle_gpu 3.2.2, python 3.11

### 🌰 Minimal Reproducible Example (最小可复现问题的Demo)

Run the eval code in the documentation
`python ./PaddleOCR/tools/eval.py -c "Short_dataset/PPOCR_config.yaml" -o Global.pretrain_model="Short_line_result/output/PP-OCRv5_server_rec/latest/latest.pdparams"`

**Comment by hjiehudbandbegshth** [comment]
Hi! As far as I can tell, there are two main reasons why your eval accuracy is 0. First, you used Global.pretrain_model in your command, but this field doesn’t exist in your config, so PaddleOCR never actually loaded your trained weights. That’s why the log says train from scratch. Since the model is randomly initialized during eval, getting 0 accuracy is expected. You should use a valid field like Global.checkpoints=xxx.pdparams or Global.pretrained_model=xxx.pdparams so the trained weights are properly loaded.

Second, your config uses a MultiHead setup (CTCHead + NRTRHead), but your post-process is CTCLabelDecode. This means eval and inference always use the CTCHead output only. If most of the learning ends up in the NRTRHead, it’s totally normal that training accuracy looks good but eval/inference outputs garbage. You either need to switch to a single CTCHead, or switch to NRTRHead and use the matching decoder. Otherwise, MultiHead won’t work correctly in deployment.
Hope this can help! I'm looking forward to your reply!

**Comment by BaoNguyen6742** [comment]
I exported the model again and the result somehow is good now
